jsonnet/kube-state-metrics.jsonnet

```diff
@@ -33,6 +33,25 @@ local tlsVolumeName = 'kube-state-metrics-tls';
                 caFile: '/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt',
                 serverName: 'server-name-replaced-at-runtime',
               },
+              // Drop the "instance" and "pod" labels since we're runinng only
+              // one instance of kube-state-metrics. The "instance" label can't
+              // be dropped at the discovery stage because otherwise Prometheus
+              // will default its value to the address being scraped.
+              // This is to avoid excessive series churn when
+              // kube-state-metrics is redeployed because of node reboot or
+              // upgrade.
+              metricRelabelings: [
```

Comment:
This also is CPU expensive as it needs to process every single series in every single scrape, do we think we are getting enough out of it, is something I feel we should consider.

Reply:
I'm going to check locally but I don't expect that it would cost us much CPU-wise.